### PR TITLE
Updates to opa gatekeeper script

### PR DESCRIPTION
**Pull request by joshualucas84**
Updates and reorg of opa gatekeeper script

### CODE REVIEW

File: platform/31-opa-gatekeeper-setup.sh [hunk 1]
```diff
@@ -1,18 +1,44 @@
...
+
+main() {
+    cleanup_cluster "@"
+    install_opa_gatekeeper "@"
+    install_secrets "@"
+}
 
-helm repo add gatekeeper https://open-policy-agent.github.io/gatekeeper/charts
-helm upgrade -f ${SCRIPT_DIR}/../resources/enable-mutating.yaml --install gatekeeper gatekeeper/gatekeeper -n gatekeeper
+main "$@"
```
Comment (bradbeck) [inline platform/31-opa-gatekeeper-setup.sh:44]: Probably want a newline at the end of the file for consistency.

Comment (joshualucas84) [reply platform/31-opa-gatekeeper-setup.sh]: Roger that.

File: platform/31-opa-gatekeeper-setup.sh [hunk 2]
```diff
@@ -1,18 +1,44 @@
...
+    kubectl delete ClusterRoleBinding gatekeeper-manager-rolebinding || true
+    kubectl delete ValidatingWebhookConfiguration gatekeeper-validating-webhook-configuration|| true
+    kubectl delete MutatingWebhookConfiguration gatekeeper-mutating-webhook-configuration || true
+    kubectl delete secret generic regcred  || true
+}
+
+
+install_opa_gatekeeper(){
+    helm repo add gatekeeper https://open-policy-agent.github.io/gatekeeper/charts
+    helm upgrade -f "${SCRIPT_DIR}"/../resources/opa-gatekeeper/enable-mutating.yaml --install gatekeeper gatekeeper/gatekeeper --version $OPA_GATEKEEPER_VERSION
```
Comment (bradbeck) [inline platform/31-opa-gatekeeper-setup.sh:27]: Does this need to have `-n gatekeeper` adde to it. Looks like the gatekeeper services are get install in the `default` namespace.

Comment (joshualucas84) [reply platform/31-opa-gatekeeper-setup.sh]: @bradbeck  thanks for pointing that out, helm should create and install everything into the gatekeeper namespace

**Review by bradbeck — APPROVED** [review]
Seems to work ok in the context of the buildpacks example.

**Comment by mlieberman85** [comment]
@joshualucas84 lgtm I think you just need to rebase

**Comment by joshualucas84** [comment]
> @joshualucas84 lgtm I think you just need to rebase

rebased

**Comment by mlieberman85** [comment]
@joshualucas84  It still says I can't merge it.

**Pull request closed without merging**